library/alignment.h

```diff
@@ -26,6 +26,11 @@
 #include <stdint.h>
 #include <string.h>
 
+#if ( defined(__ARMCC_VERSION) || defined(_MSC_VER) ) && \
+     !defined(inline) && !defined(__cplusplus)
+#define inline __inline
+#endif
```

Comment:
we should really have a common header file that is always included, with things like this in

Reply:
@gilles-peskine-arm does exactly this in #6648 - so if that goes in first I'll update this PR

Reply:
I've merged in this change and removed the redundant inline definitions.